performant-software/DM
# Recover scroll position on text editable toggle, highlight text targets consistently with images

### What does this PR do?
- adds a behavior that, when a text editor is switched between editable and uneditable, returns the scrolling editor pane to the same point in the text rather than to the start
- makes text-target highlighting consistent with how image targets are highlighted: when a text resource is opened from an annotation link, the resource will at first show only the highlight attached to the annotation (other highlights will be shown if the show-highlights toggle is turned off then on, as with image markers)
- improves how text highlight visibility is maintained, avoiding modifications of text that could get retained in the backend if the browser window is closed

### What issues does this address?
addresses #63, #92 

### How to test
On staging (http://67.207.158.85/), open a variety of text resources from both the search/TOC and from annotation links in the rollover window. Confirm that the latter only show the single targeted highlight at first but will show others when highlight display is toggled off and on. Toggle text resources in and out of editing mode while scrolled to various points in longer text resources, confirming that the scroll position stays roughly (font change will always throw it off a little bit) the same.
